sdk/core/azure_core_amqp/src/fe2o3/connection.rs

```diff
@@ -52,10 +52,24 @@ impl AmqpConnectionApis for Fe2o3AmqpConnection {
                 .max_frame_size(65536);
 
             if options.is_some() {
-                let options = options.unwrap();
+                let options = options.ok_or_else(|| {
+                    azure_core::Error::new(
+                        azure_core::error::ErrorKind::Other,
+                        "Connection options are not set.",
+                    )
+                })?;
+
+                // options.max_frame_size().inspect(move |s| {
+                //     builder.max_frame_size(*s);
+                // });
                 if options.max_frame_size.is_some() {
                     builder = builder.max_frame_size(options.max_frame_size.unwrap());
                 }
```

Comment:
As discussed offline:
```suggestion
                if let Some(max_frame_size) = options.max_frame_size {
                    builder = builder.max_frame_size(max_frame_size);
                }
```

Use `as_ref()` for non-`Copy`ables.